src/routes/user/userRoute.js

```diff
@@ -18,6 +18,10 @@ const userRoutes = (router) => {
     res.clearCookie('auth');
     return res.redirect('/api/');
   });
+
+  router.patch('/users/:id', auth, async (req, res) => {
+    userDeletion(req, res);
```

Comment:
As I mentioned above req and res should stay in routes and not be passed to the controller. You shoul pass only id.